Client.Core/Flux/Internal/FluxCsvParser.cs

```diff
@@ -72,119 +76,154 @@ public void ParseFluxResponse(Stream source, ICancellable cancellable, IFluxResp
-                    if ("#datatype".Equals(token))
-                    {
-                        AddDataTypes(table, csv);
-                    }
-                    else if ("#group".Equals(token))
-                    {
-                        AddGroups(table, csv);
-                    }
-                    else if ("#default".Equals(token))
-                    {
-                        AddDefaultEmptyValues(table, csv);
-                    }
+                    if (record == null)
+                        consumer.Accept(state.tableIndex, cancellable, table);
                     else
-                    {
-                        // parse column names
-                        if (startNewTable)
-                        {
-                            AddColumnNamesAndTags(table, csv);
-                            startNewTable = false;
-                            continue;
-                        }
-
-                        int currentId;
-
-                        try
-                        {
-                            currentId = Convert.ToInt32(csv[1 + 1]);
-                        }
-                        catch (Exception)
-                        {
-                            throw new FluxCsvParserException("Unable to parse CSV response.");
-                        }
-                        if (tableId == -1) {
-                            tableId = currentId;
-                        }
-
-                        if (tableId != currentId)
-                        {
-                            //create new table with previous column headers settings
-                            var fluxColumns = table.Columns;
-                            table = new FluxTable();
-                            table.Columns.AddRange(fluxColumns);
-                            consumer.Accept(tableIndex, cancellable, table);
-                            tableIndex++;
-                            tableId = currentId;
-                        }
-
-                        var fluxRecord = ParseRecord(tableIndex - 1, table, csv);
-                        consumer.Accept(tableIndex - 1, cancellable, fluxRecord);
-                    }
+                        consumer.Accept(state.tableIndex - 1, cancellable, record);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Parse Flux CSV response to <see cref="IAsyncEnumerable{T}"/>.
+        /// </summary>
+        /// <param name="reader">CSV Data source reader</param>
+        /// <param name="cancellationToken">cancellation token</param>
+        public async IAsyncEnumerable<(FluxTable, FluxRecord)> ParseFluxResponseAsync(StringReader reader, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            Arguments.CheckNotNull(reader, nameof(reader));
+
+            using var csv = new CsvReader(reader, CultureInfo.InvariantCulture);
+            ParseFluxResponseState state = new ParseFluxResponseState { csv = csv };
```

Comment:
Use 'var' (simple types):

```suggestion
            var state = new ParseFluxResponseState { csv = csv };
```